design-proposals/passt-feature-gate-iface-api-deprecation.md

```diff
@@ -0,0 +1,66 @@
+## Overview
+Following the emerging [Kubevirt Network Binding Plugin API](https://docs.google.com/document/d/13fLaqZuM8El_CT5XzxPv-1epKsB1-Spv_jtXveEkkdQ/edit#heading=h.z27rudfpwy44)
+and introduction of Passt network binding plugin [2](https://github.com/kubevirt/kubevirt/pull/10425) [3](https://github.com/kubevirt/kubevirt/pull/10621), 
+it's now possible to remove Passt network binding implementation from Kubevirt core.
+
+This doc outlines deprecation plan for Passt network binding, interface API and  feature gate.
+
+# Motivation
+Removing Passt network binding implementation from Kubevirt core and use Passt network binding plugin instead, 
+enable reduce the maintenance burden.
+Having Passt binding code outside Kubevirt's core may ease contribution process as there is no need to be familiar 
+with [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt) core code.
+
+## Goals
+- Deprecate Passt network interface API.
+- Remove Passt feature gate.
+- Remove Passt network binding from Kubevirt core.
+  
+## Non Goals
+
+## User Stories
+
+## Repos
+[kubevirt/kubevirt](https://github.com/kubevirt/kubevirt):
+- [Passt binding plugin hook sidecar](https://github.com/kubevirt/kubevirt/tree/main/cmd/sidecars/network-passt-binding)
+- [Passt binding plugin CNI](https://github.com/kubevirt/kubevirt/tree/main/cmd/cniplugins/passt-binding)
+
+# Design
+Deprecating Passt feature-gate and interface API and replacing it with Passt binding plugin
+require some changes users should be aware of and prepared for.
+
+The proposed deprecation plan for Passt binding interface API and feature-gate is as follows:
+
+1. Announce passt feature-gate and interface API deprecation in kubevirt-dev mailing list.
+   This will give users some time to prepare for changes in Kubevirt next release v1.2.0. 
+
+2. Kubevirt next release v1.2.0 should have the following changes: 
+   1. VM whose spec specifies passt interface (`spec.domain.devices.interface.[].passt`) will be created with 
+      Passt network binding hook sidecar and CNI, in a way Kubevirt core implementation is not executed.
+
+      This will require users to set the following Passt binding plugin prequisites in advance:
+      1.  Install Passt binding CNI plugin in cluster nodes
+      2. Create and register Passt binding hook sidecar image and network-attachment-definition, similar to 
+         [Kubevirt Network Binding Plugin - bridge binding example](https://docs.google.com/document/d/13fLaqZuM8El_CT5XzxPv-1epKsB1-Spv_jtXveEkkdQ/edit#heading=h.z27rudfpwy44).
+
+      If one or more is missing, newly created VMs with Passt interface won't be able to start, existing ones will fail migrate.
+
+   2. [Passt interface API](http://kubevirt.io/api-reference/main/definitions.html#_v1_interfacepasst) 
```

Comment:
I don't understand how point 1 and 2 can live together? They have two possible solution to the same API.
I would keeo just point 2.

Reply:
Sorry for the mess, fixed.